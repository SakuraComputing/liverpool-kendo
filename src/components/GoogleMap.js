import React from 'react';

import GoogleMapReact from 'google-map-react';

const AnyReactComponent = ({ text }) => <div>{text}</div>;

export class SimpleMap extends React.Component {
    static defaultProps = {
        center: {lat: 53.3701668, lng: -2.9195765},
        zoom: 15
    };

    render() {
        return (
            <GoogleMapReact
                bootstrapURLKeys={{ key: 'AIzaSyADAT64SMWOP7ee4cEOgW8tMYNL673MeDQ' }}
                defaultCenter={this.props.center}
                defaultZoom={this.props.zoom}
            >
                <AnyReactComponent
                    lat={53.3701668}
                    lng={-2.9195765}
                    text={'Liverpool Kendo Club'}
                />
            </GoogleMapReact>
        );
    }
}
export default SimpleMap;
