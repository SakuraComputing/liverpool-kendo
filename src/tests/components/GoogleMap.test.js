import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import GoogleMap from '../../components/GoogleMap';

test('should render the Google Map component',() => {
    const wrapper = shallow(<GoogleMap />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});
