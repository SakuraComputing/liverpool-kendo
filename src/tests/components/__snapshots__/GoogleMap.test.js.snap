// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Google Map component 1`] = `
<GoogleMap
  bootstrapURLKeys={
    Object {
      "key": "AIzaSyADAT64SMWOP7ee4cEOgW8tMYNL673MeDQ",
    }
  }
  debounced={true}
  defaultCenter={
    Object {
      "lat": 53.3707973,
      "lng": -2.9188403,
    }
  }
  defaultZoom={16}
  distanceToMouse={[Function]}
  googleMapLoader={[Function]}
  heatmap={Object {}}
  heatmapLibrary={false}
  hoverDistance={30}
  layerTypes={Array []}
  options={[Function]}
  style={
    Object {
      "height": "100%",
      "margin": 0,
      "padding": 0,
      "position": "relative",
      "width": "100%",
    }
  }
  yesIWantToUseGoogleMapApiInternals={false}
>
  <AnyReactComponent
    lat={53.3707973}
    lng={-2.9188403}
  />
</GoogleMap>
`;
